### PR TITLE
Add `PromiseKit` 6.1.1. to example project [Option E]

### PR DESCRIPTION
**Pull request by bachand**
The PR is the simplest solution for getting `PromiseKit` 6.1.1. integrated into `BuckSample`.

The issue we were having was the Buck-generated project wasn't building after we added `-all_load`. We were getting duplicate symbol errors.

It makes sense now why this was happening. Our Buck definition for `PromiseKit` was including too many files to compile.

The `PromiseKit` podspec doesn't list some files as sources.
Screen Shot 2019-03-20 at 1 25 32 PM

Here are the source being compiled by our manually managed project, which consumes `PromiseKit` via the podspec.
Screen Shot 2019-03-20 at 1 26 26 PM

Excluding two files gets the project to build.

Our manually managed project has 27 sources but this project has 26. The only file missing is `PromiseKit-dummy.m`, which I suspect is added by CocoaPods.

This is a replace for https://github.com/airbnb/BuckSample/pull/55 and https://github.com/airbnb/BuckSample/pull/56

### CODE REVIEW

File: Pods/BUCK [hunk 1]
```diff
@@ -17,6 +17,26 @@ apple_third_party_lib(
...
 
+apple_third_party_lib(
+    name = "PromiseKit",
+    visibility = ["PUBLIC"],
+    exported_headers = glob([
+        "PromiseKit/**/*.h",
+    ], exclude = [
+        "PromiseKit/AnyPromise+Private.h",
+    ]),
+    headers = glob([
```
Comment (dfed) [inline Pods/BUCK]: Since the `podspec` does not include any private headers, I don't think we need this line

File: Pods/BUCK [hunk 2]
```diff
@@ -17,6 +17,26 @@ apple_third_party_lib(
...
+        "PromiseKit/**/*.h",
+    ]),
+    srcs = glob([
+        "PromiseKit/**/*.m",
+        "PromiseKit/**/*.swift",
+    ], exclude = [
```
Comment (dfed) [inline Pods/BUCK]: 👏

File: Pods/BUCK (hunk 1 above)
Comment (bachand) [reply Pods/BUCK]: Good call. Removed.

File: Podfile [hunk 3]
```diff
@@ -11,6 +11,8 @@ target 'BuckSample' do
   pod 'CryptoSwift'
   pod 'OHHTTPStubs'
   pod 'Bugsnag'
+  pod 'PromiseKit/CorePromise', '6.1.1'
```
Comment (bachand) [inline Podfile]: There is nothing special about this version. It's just what we're using currently.

**Comment by bachand** [comment]
I added code to use `PromiseKit` in `ViewController.swift` and verified that it works.

Screen Shot 2019-03-20 at 3 33 46 PM

File: Pods/BUCK (hunk 2 above)
Comment (dfed) [reply Pods/BUCK]: A comment on why we're excluding these things would be dope.

Comment (bachand) [reply Pods/BUCK]: Great idea: https://github.com/airbnb/BuckSample/pull/58/commits/30bdd930fb85c282ed404027579be0959b2fce1d

**Pull request merged**